### PR TITLE
DAOS-11866 build: Add --build-deps=fetch option

### PR DESCRIPTION
**Pull request by jolivier23**
Adds --skip-download option to allow one to use already download sources for prerequisites.
Removes --build-deps=build-only as it can be replaced with --build-deps=only --skip-download
Adds --build-deps=fetch which just downloads sources and exits.
A few minor cleanup changes.

Required-githooks: true

Change-Id: Icf0023b66ea8f4bd02fe054b8509bc10a1c6c957

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by jolivier23** [comment]
@daltonbohning is this useful to you?  It was requested in the ticket, basically a way to download without building

File: site_scons/prereq_tools/base.py [hunk 1]
```diff
@@ -675,11 +686,13 @@ def save_build_info(self):
     def __parse_build_deps(self):
         """Parse the build dependencies command line flag"""
         build_deps = GetOption('build_deps')
-        if build_deps in ('yes', 'only'):
-            self.download_deps = True
-            self.build_deps = True
-        elif build_deps == 'build-only':
+        skip_download = GetOption('skip_download')
+        if build_deps in ('fetch'):
```
Comment (daltonbohning) [inline site_scons/prereq_tools/base.py:690]: minor - this syntax actually doesn't do what it looks like
A single string in parens evaulates to a string. Whereas a single string in parens, with a comma, evaluates to a `tuple` with a single element
```
>>> ('fetch')
'fetch'
>>> ('fetch',)
('fetch',)
```
So notice both of these are `True`
```
>>> 'fetch' in ('fetch')
True
>>> 'fetch' in ('fetchplus')
True
```
but not this way
```
>>> 'fetch' in ('fetch',)
True
>>> 'fetch' in ('fetchplus',)
False
```

**Comment by daltonbohning** [comment]
> @daltonbohning is this useful to you? It was requested in the ticket, basically a way to download without building

I don't necessarily need this, but I could see it as being useful for flows like
1. Checkout deps
2. Manually make some deps changes
3. Build

**Comment by jolivier23** [comment]
> > @daltonbohning is this useful to you? It was requested in the ticket, basically a way to download without building
> 
> I don't necessarily need this, but I could see it as being useful for flows like
> 
> 1. Checkout deps
> 2. Manually make some deps changes
> 3. Build

Ok, was wondering if it could simplify your frontera testing too but perhaps not. Original ticket is from @mchaarawi

**Comment by daltonbohning** [comment]
> Ok, was wondering if it could simplify your frontera testing too but perhaps not. Original ticket is from @mchaarawi

On Frontera I build on a login node that has internet access. But that workflow I mentioned has come up before

**Review by daltonbohning — APPROVED** [review]

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**